### PR TITLE
Tournaments: Refactor commands

### PR DESCRIPTION
**Pull request by mia-pi-git**
ಠ_ಠ
(this is so perms can be toggled properly and so metadata can be properly accessed)

### CODE REVIEW

File: server/tournaments/index.ts [hunk 1]
```diff
@@ -1232,21 +1435,26 @@ const tourCommands: {basic: TourCommands, creation: TourCommands, moderation: To
...
-		settype(tournament, user, params, cmd) {
-			if (params.length < 1) {
-				return this.sendReply(`Usage: ${cmd} <type> [, <comma-separated arguments>]`);
+		settype(target, room, user, connection, cmd) {
+			if (!room) return this.requiresRoom();
+			const tournament = room.getGame(Tournament);
+			if (!tournament) return this.errorReply(`There is no tournament running.`);
+			const [cap, name, modifier] = target.split(',').map(item => item.trim());
+			if (!target) {
+				return this.sendReply(`Usage: /tour ${cmd} <type> [, <comma-separated arguments>]`);
```
Comment (AnnikaCodes) [inline server/tournaments/index.ts:1454]: Shouldn't you be checking if `target` exists _before_ calling split?

File: server/tournaments/index.ts [hunk 2]
```diff
@@ -1377,56 +1603,66 @@ const tourCommands: {basic: TourCommands, creation: TourCommands, moderation: To
...
+			if (!room) return this.requiresRoom();
+			const tournament = room.getGame(Tournament);
+			if (!tournament) return this.errorReply(`There is no tournament running.`);
+			if (!target) {
+				return this.sendReply(`Usage: /tour ${cmd} <user>`);
 			}
-			const targetUser = Users.get(params[0]);
+			const [userid, reason] = target.split(',').map(item => item.trim());
+			const targetUser = Users.get(userid);
 			const targetUserid = toID(targetUser);
```
Comment (AnnikaCodes) [inline server/tournaments/index.ts:1636]: `targetUser` can be null here so if you want to support disqualifying nonexistent User objects, you might want a fallback for `targetUserid`

**Comment by whalemer** [comment]
is this just so that you can use the subcommand metadata properly?

**Comment by Zarel** [comment]
Yes, @whalemer, I believe so.

**Comment by KrisXV** [comment]
bumping this because the roa room would benefit from this in /permissions

**Comment by Zarel** [comment]
I'm definitely in favor of this; @mia-pi-git can you rebase or merge master?

**Review by Zarel — APPROVED** [review]

**Comment by mia-pi-git** [comment]
Done, @Zarel.

**Comment by Zarel** [comment]
Looks good, hopefully no bugs.

**Pull request merged**